packages/sdk/observability/src/segment/types.ts

```diff
@@ -2,26 +2,48 @@
 // Copyright 2022 DXOS.org
 //
 
+export type Tags = Record<string, string>;
+
 export type SegmentTelemetryOptions = {
   apiKey?: string;
   batchSize?: number;
   enable?: boolean;
-  getTags: () => { [key: string]: string };
+  getTags: () => Tags;
 };
 
-type CommonOptions = {
-  installationId?: string;
-  did?: string;
-  properties?: object;
+// Copied from @segment/analytics-node.
+export type SegmentIdentityOptions =
+  | { userId: string; anonymousId?: string }
+  | { userId?: string; anonymousId: string };
+
+export type IdentityOptions = { did: string; installationId?: string } | { did?: string; installationId: string };
+
+export type CommonOptions = IdentityOptions & {
   timestamp?: Date;
+  properties?: Record<string, unknown>; // TODO(burdon): Rename custom.
 };
 
+/**
+ * Page views.
+ * https://segment.com/docs/connections/sources/catalog/libraries/server/node/#page
+ */
 export type PageOptions = CommonOptions & {
-  installationId?: string;
-  name?: string;
   category?: string;
+  name: string;
 };
 
-export type EventOptions = CommonOptions & {
-  name: string;
+/**
+ * Track actions.
+ * https://segment.com/docs/connections/sources/catalog/libraries/server/node/#track
+ */
+export type TrackOptions = CommonOptions & {
+  event?: Event;
+  action: string;
 };
+
+/**
+ * Each event will be mapped to a different Postgres table via segment.
+ */
+export enum Event {
+  USER = 'user',
```

Comment:
I would say that this event https://github.com/dxos/dxos/pull/8911/files#diff-d457dd0159c665b772f2e7a6a8e0697e212cb9f4c0f8f15ab3ff80f3bb4abc59R410-R414 is not a user event, however iirc its also super spammy, so we maybe want to disable it anyways and consider another way to collect that